apps/api/src/auth/auth.service.ts

```diff
@@ -145,6 +145,22 @@ export class AuthService {
       if (!grant.access_token?.token) {
         throw new InternalServerErrorException('CannotIssueTokenError')
       }
+      const user = await this.keycloak.grantManager.userInfo<string, KeycloakTokenParsed>(
+        grant.access_token.token as string,
+      )
+      const person = await this.prismaService.person.findUnique({ where: { email: user.email } })
+      if (!person || person.keycloakId !== user.sub) {
+        Logger.warn('No person found for the current keycloak user. Creating new one...')
+        await this.authenticateAdmin()
+        const userData = await this.admin.users.findOne({ id: user.sub })
```

Comment:
What if there is no user? I know it is highly unlikely, but maybe we should throw an error...

Reply:
If there is no user in the keycloak server, login will be unsuccessful.